src/reporters/LambdaReporter.ts

```diff
@@ -0,0 +1,104 @@
+import { HttpClient } from "@actions/http-client";
+import { throws } from "assert";
+import { CheckOptions } from "../check";
+import { Result } from "../ComplianceChecker";
+import { Product } from "../model/Product";
+import { Reporter } from "./Reporter";
+
+interface ResultRecord {
+  level: string,
+  rule?: string,
+  check?: string,
+  mandatory?: string,
+  result: string,
+  reason?: string
+}
+
+interface RequestBody {
+  repo_owner?: string,
+  repo_name?: string,
+  start_time?: string,
+  end_time?: string,
+  result?: ResultRecord[]
+}
+
+export class LambdaReporter extends Reporter {
+
+  private readonly uri: string;
+  private readonly keyHeader: any;
+  private readonly http: HttpClient;
+  private body: RequestBody;
+
+  constructor() {
+    super();
+    if (process.env.INPUT_ENG_COMPLIANCE_LAMBDA_URI == null) throw new Error('INPUT_ENG_COMPLIANCE_LAMBDA_URI must be specified');
```

Comment:
<details>
<summary>:warning: <strong>This code contains single line control structures without curly braces, `{}`.</strong> (click for details)</summary>

Consider surrounding these statements with `{ }`.  
  
Braces are used around all statements, even singletons, when they are part of a control structure, such as an `if`-`else` or `for` statement.  
  
This makes it easier to add statements without accidentally introducing bugs due to forgetting to add braces. Readability is also significantly improved.

**[More details](https://confluence.devfactory.com/x/fHvlFQ)**

## The improvement suggestion in this case
```suggestion  
    if (process.env.INPUT_ENG_COMPLIANCE_LAMBDA_URI == null) {
      throw new Error('INPUT_ENG_COMPLIANCE_LAMBDA_URI must be specified');
    }
```  
> NOTE: Consider applying the structure of this solution in other parts of the code, if there is another occurrence(s).
</details>

[//]: # (<a id="30ed0ecf-3cfd-4b0f-ae33-0394e39742c8" data-finding-id="367360892" href="none" codereview-uuid-reserved-tag data-business-id="55ce652a-40cf-48e1-916a-f8d32a3b8762" data-analyzer="js" />)